### ISSUE TITLE
Object properties without an explicit reference specifier are declared as assign

### ISSUE THREAD

**Issue by steipete** [realm/jazzy#829]
Properties that do not have a ‘strong’ or ‘copy’ on them in source are exported as ‘assign’ in the documentation. We omit these modifiers as they are inferred automatically and the property syntax is already long enough with nullability added.

For example, this is declared in code:

```
/// Stamp image. Defaults to nil. Set to render an image.
/// @note An image set will take priority over the internal appearance stream.
@property (nonatomic, nullable) UIImage *image;
```

This is what we get:

```
@property (assign, readwrite, nonatomic, nullable) UIImage *image;
```

https://pspdfkit.com/api/ios/Classes/PSPDFStampAnnotation.html#/c:objc(cs)PSPDFStampAnnotation(py)image

It's been on our back burner for a while to fix, but haven't found time so far. Do you have a starting point for me how this can e fixed?

**Comment by johnfairh** [comment]
The extra attributes are introduced by sourcekitten [here](https://github.com/jpsim/SourceKitten/blob/c01833de782d2e8866f3ada33ed266aa585c691a/Source/SourceKittenFramework/Clang%2BSourceKitten.swift#L87) -- they show up only if the decl has a doc comment which gives the code access to the fuller version of the declaration that the toolchain knows, rather than what has been parsed from the source file.

The Swift version stores both what the user typed (`parsed_declaration`) and what the tools know (`doc.declaration`) and jazzy prefers the former for display.  Maybe SK should do the same for ObjC?

**Comment by jpsim** [comment]
Yeah, we get those from libclang. Maybe there's an extra flag we can pass it to help it better assume what default property attributes to add?

**Comment by johnfairh** [comment]
The libclang in Xcode 11.4 makes this worse by also adding `unsafe_retained` [(commit)](https://reviews.llvm.org/D57965), so the line above comes out
```
@property (nonatomic, assign, unsafe_unretained, readwrite, nullable) UIImage *image;
```

So, tentatively propose to strip out the default property attributes `atomic`, `readwrite`, `assign`, `unsafe_retained` -- this *seems* to be how apple docs work (or they could have a stricter in-house style about not writing them.)

Could do with an opinion from genuine ObjC developers: is this OK or is there a legitimate reason to preserve them if written/show them?  I guess would have a jazzy option to disable property attribute stripping.

(There is no way to persuade libclang not to do this -- [code](https://github.com/llvm/llvm-project/blob/93184a8eda272c65308906836b47cbf209de779e/clang/lib/AST/DeclPrinter.cpp#L1422).)

**Comment by johnfairh** [comment]
@steipete @jpsim @joesus any opinions on this, if you're still working in Objective-C?

**Comment by steipete** [comment]
Yes, we still have a large customer base that has at least partial ObjC codebases, so this requirement won't go away for a long time.

>strip out the default property attributes atomic, readwrite, assign, unsafe_retained

This sounds like the best way forward; cleans up the declaration without removing any information that can't be inferred. Most ObjC style-guides define that these are fluff and should be avoided, especially with generics declarations are already long enough. I'm not even sure a property for this is needed.

**Comment by joesus** [comment]
Agreed. Additionally it's my opinion that any developer sophisticated enough to care about the nuances of atomic vs nonatomic will be a developer that has no issue finding the declaration in the source code. Thanks as always for the great work!!!

**Comment by johnfairh** [comment]
Thanks for the feedback.  Will include this in the Xcode 11.4 patches.

**Comment by johnfairh** [comment]
Fixed in master via #1158.
